File: python/src/omnifocus_mcp/resources.py
from omnifocus_mcp.registration import typed_resource
from omnifocus_mcp.app import mcp
from omnifocus_mcp.tools.forecast import get_forecast
from omnifocus_mcp.tools.projects import list_projects
from omnifocus_mcp.tools.tasks import get_inbox


@typed_resource(mcp, "omnifocus://inbox")
async def inbox_resource() -> str:
    """resource for current inbox tasks as json."""
    return await get_inbox()


@typed_resource(mcp, "omnifocus://today")
async def today_resource() -> str:
    """resource for forecast sections as json."""
    return await get_forecast()


@typed_resource(mcp, "omnifocus://projects")
async def projects_resource() -> str:
    """resource for active project summaries as json."""
    return await list_projects(status="active")

File: rust/src/resources.rs
use crate::{
    error::Result,
    jxa::JxaRunner,
    tools::{forecast::get_forecast, projects::list_projects, tasks::get_inbox},
};

pub const INBOX_RESOURCE_URI: &str = "omnifocus://inbox";
pub const TODAY_RESOURCE_URI: &str = "omnifocus://today";
pub const PROJECTS_RESOURCE_URI: &str = "omnifocus://projects";

pub async fn inbox_resource<R: JxaRunner>(runner: &R) -> Result<String> {
    let tasks = get_inbox(runner, 100).await?;
    Ok(serde_json::to_string(&tasks)?)
}

pub async fn today_resource<R: JxaRunner>(runner: &R) -> Result<String> {
    let forecast = get_forecast(runner, 100).await?;
    Ok(serde_json::to_string(&forecast)?)
}

pub async fn projects_resource<R: JxaRunner>(runner: &R) -> Result<String> {
    let projects = list_projects(runner, None, "active", None, None, false, None, "asc", 100).await?;
    Ok(serde_json::to_string(&projects)?)
}
